src/scaladoc/scala/tools/nsc/doc/html/page/JSON.scala

```diff
@@ -0,0 +1,71 @@
+package scala.tools.nsc.doc.html.page
+
+import JSONFormat.format
+
+object JSONFormat {
+
+  /**
+   * The default formatter used by the library. You can
+   * provide your own with the toString calls on
+   * JSONObject and JSONArray instances.
+   */
+  def format(x: Any): String = x match {
+    case s:  String     => s"""\"${quoteString(s)}\""""
+    case jo: JSONObject => jo.toString
+    case ja: JSONArray  => ja.toString
+    case other          => other.toString
+  }
+
+  /**
+   * This function can be used to properly quote Strings
+   * for JSON output.
+   */
+  def quoteString (s: String) : String = {
+    val buf = new StringBuilder(s.length)
+    val len: Int = s.length
+    var i: Int = 0
+    while (i < len) {
+      s.apply(i) match {
+        case '"'  => buf ++= "\\\""
+        case '\\' => buf ++= "\\\\"
+        case '/'  => buf ++= "\\/"
+        case '\b' => buf ++= "\\b"
+        case '\f' => buf ++= "\\f"
+        case '\n' => buf ++= "\\n"
+        case '\r' => buf ++= "\\r"
+        case '\t' => buf ++= "\\t"
+        /* We'll unicode escape any control characters. These include:
+         * 0x0 -> 0x1f  : ASCII Control (C0 Control Codes)
+         * 0x7f         : ASCII DELETE
+         * 0x80 -> 0x9f : C1 Control Codes
+         *
+         * Per RFC4627, section 2.5, we're not technically required to
+         * encode the C1 codes, but we do to be safe.
+         */
+        case c if ((c >= '\u0000' && c <= '\u001f') || (c >= '\u007f' && c <= '\u009f')) =>
+                     val cint = c.toInt
+                     buf ++= f"\\u$cint%04x"
+        case c    => buf += c
+      }
+      i += 1
+    }
+    buf.toString()
+  }
+}
+
+/**
+ *  Represents a JSON Object (map).
+ *
+ *  @author Derek Chen-Becker <"java"+@+"chen-becker"+"."+"org">
+ */
+case class JSONObject (obj: Map[String,Any]) {
+  override def toString = s"{${obj.map({ case (k,v) => format(k) + " : " + format(v) }).mkString(", ")}}"
```

Comment:
`obj.map { case (k,v) => s"${format(k)} : ${format(v)}" }.mkString("{", ", ", "}")` to omit extra parens and use `s` inside.